core/src/main/java/com/linecorp/armeria/internal/common/util/TemporaryThreadLocals.java

```diff
@@ -15,82 +15,124 @@
  */
 package com.linecorp.armeria.internal.common.util;
 
+import static com.google.common.base.Preconditions.checkState;
+
 import com.google.common.annotations.VisibleForTesting;
 
 import io.netty.util.internal.EmptyArrays;
 
 /**
  * Provides various thread-local variables used by Armeria internally, mostly to avoid allocating
  * short-living objects, such as {@code byte[]} and {@link StringBuilder}. Keep in mind that the variables
- * provided by this class must be used with extreme care, because otherwise it will result in unpredictable
- * behavior.
+ * provided by this class must be used and released with extreme care, because otherwise it will result in
+ * unpredictable behavior.
  *
  * <p>Most common mistake is to call or recurse info a method that uses the same thread-local variable.
  * For example, the following code will produce a garbled string:
  * <pre>{@code
  * > class A {
  * >     @Override
  * >     public String toString() {
- * >         return TemporaryThreadLocals.get().append('"').append(new B()).append('"').toString();
+ * >         TemporaryThreadLocals tempThreadLocals = TemporaryThreadLocals.acquire();
+ * >         StringBuilder stringBuilder = tempThreadLocals.stringBuilder();
+ * >         tempThreadLocals.close();
+ * >         return stringBuilder.append('"').append(new B()).append('"').toString();
  * >     }
  * > }
  * > class B {
  * >     @Override
  * >     public String toString() {
- * >         return TemporaryThreadLocals.get().append("foo").toString();
+ * >         StringBuilder stringBuilder = TemporaryThreadLocals.acquire().stringBuilder();
+ * >         return stringBuilder.append("foo").toString();
  * >     }
  * > }
  * > // The following assertion fails, because A.toString() returns "foofoo\"".
  * > assert "\"foo\"".equals(new A().toString());
- * }</pre></p>
+ * }</pre>
```

Comment:
How about removing this legacy API usages and using only try-with-resouce for the wrong used case.
It would be clear for users to provide a correct example using try-with-resources for less confusion.
If a method returns `AutoCloseable`, it also implies to must close the resource.

Reply:
Sounds good. Revised. Please take a look again 🙏